internal/gittuf-git/push/push.go

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package push
+
+import (
+	"os"
+	"os/exec"
+
+	"github.com/gittuf/gittuf/internal/gittuf-git/args"
+)
+
+// GittufPush handles the gittuf push operation.
+func GittufPush(gitArgs args.Args) error {
+	// 0. Determine appropriate env vars
+	// 1. Record changes to the RSL
+	// 2. Push non-RSL changes to the remote
+	// 3. Push RSL changes to the remote
+
+	var env = os.Environ()
+
+	// 0
+	if gitArgs.ChdirIdx > 0 {
+		env = append(env, "GIT_DIR="+gitArgs.GlobalFlags[gitArgs.ChdirIdx])
+	}
+
+	// 1
+	rslRecordCmd := exec.Command("gittuf", "rsl", "record")
```

Comment:
Is this recording HEAD? The push may be for a completely different branch than what's checked out. Unfortunately, you'd have to also evaluate what refs are being pushed (and also process refspecs if that's passed in).